src/main/java/duke/DeleteCommand.java

```diff
@@ -0,0 +1,44 @@
+package duke;
+
+/**
+ * Encapsulates a command to delete a task from list of tasks.
+ */
+public class DeleteCommand implements Command {
+    private int deleteIndex;
+
+    /**
+     * Creates a new command to delete a task from list of tasks.
+     *
+     * @param deleteIndex the index of the task to be deleted.
+     */
+    public DeleteCommand(int deleteIndex) {
+        this.deleteIndex = deleteIndex;
+    }
+
+    /**
+     * Deletes the task corresponding to the deleteIndex in the task list provided.
+     *
+     * @param tasks the current list of tasks.
+     * @param ui user interface interacts with the user.
+     * @param storage custodian of reading and writing save files.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        if (deleteIndex < 0 || deleteIndex >= tasks.size()) {
+            throw new DukeException("Task with that task number does not exist.");
+        }
+        Task deletedTask = tasks.delete(deleteIndex);
+        ui.announceTaskDeletion(deletedTask, tasks.size());
+        storage.save(tasks);
+    }
```

Comment:
Perhaps you may want to add `throws DukeException` on the method. This helps to identify what kind of exceptions that are thrown by a method.